### PR TITLE
feat: ITO lazy load history list

### PR DESCRIPTION
**Pull request by zhouhanseng**
closes https://github.com/DimensionDev/Maskbook/issues/3414

https://user-images.githubusercontent.com/63733714/123776126-943aac80-d901-11eb-81c8-ff05c37c9145.mov

### CODE REVIEW

File: packages/maskbook/src/plugins/ITO/UI/PoolList.tsx [hunk 1]
```diff
@@ -34,7 +34,10 @@ export function PoolList(props: PoolListProps) {
     const { value: pools = [], loading, retry } = useAllPoolsAsSeller(account, page)
 
     const containerRef = useRef<HTMLDivElement>(null)
-    const addPage = useCallback(() => setPage(page + 1), [page])
+    const addPage = useCallback(() => {
+        console.log({ page })
```
Comment (septs) [inline packages/maskbook/src/plugins/ITO/UI/PoolList.tsx:38]: remove log?

File: packages/shared/src/hooks/useScrollBottomEvent.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useLayoutEffect, RefObject, useCallback } from 'react'
+import { debounce } from 'lodash-es'
+
+export function useScrollBottomEvent(ref: RefObject<HTMLDivElement>, cb: () => void) {
```
Comment (Jack-Works) [inline packages/shared/src/hooks/useScrollBottomEvent.ts:4]: I recommend using https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserver.

Comment (zhouhanseng) [reply packages/shared/src/hooks/useScrollBottomEvent.ts:4]: Append an element at the bottom of the container, as it appears trigger the event. Maybe this is a little bit complex than the current implementation.

**Comment by AinuX** [comment]
Test passed with build (29769)

File: packages/maskbook/src/plugins/ITO/UI/PoolList.tsx (hunk 1 above)
Comment (zhouhanseng) [reply packages/maskbook/src/plugins/ITO/UI/PoolList.tsx]: removed

**Review by septs — APPROVED** [review]

**Review by nuanyang233 — APPROVED** [review]

File: packages/shared/src/hooks/useScrollBottomEvent.ts (hunk 2 above)
Comment (Jack-Works) [reply packages/shared/src/hooks/useScrollBottomEvent.ts:4]: To use IntersectionObserver, we also add a new element at the bottom and observe its distance to the viewport. It's better than watching `scroll` event and calculate the distance by yourself.

You can use this: https://github.com/streamich/react-use/blob/master/docs/useIntersection.md

**Pull request merged**